qcore/utils.py

```diff
@@ -123,20 +123,36 @@ def load_sim_params(sim_yaml_path, load_fault=True, load_root=True, load_vm=True
     :param load_vm: to load vm_params.yaml or not
     :return: a DotDictify object that contains all necessary params for a single simulation
     """
-    sim_params = load_yaml(sim_yaml_path)
+    sim_params = {}
     fault_params = {}
     root_params = {}
     vm_params = {}
-    if load_root or load_vm and not load_fault:
+
+    if load_root is True or load_vm is True and not load_fault:
         load_fault = True  # root/vm_yamlpath in fault_yaml
-    if load_fault:
+
+    if sim_yaml_path:
```

Comment:
Can we do all the new smart & direct load_yaml when this if test fails, instead of having them spread over?

Reply:
Do you mean, have the old style loading for when everything is available, and have a separate/new style method of loading for when there are only partial params files available?
I think this is neater/more concise than having two methods, but I'm open to input if others are onboard with it

Reply:
I meant, something like this. 
```
if sim_yaml_path: # which is when load_fault, load_root, load_vm are boolean
    sim_params = load_yaml(sim_yaml_path)
    if load_fault:
        fault_params = load_yaml(sim_params["fault_yaml_path"])
    if load_root:                
       root_params = load_yaml(fault_params["root_yaml_path"])
    if load_vm:       
       vm_params = load_yaml(
            os.path.join(fault_params["vel_mod_dir"], "vm_params.yaml")
        )
 else: # when load_fault, load_root, load_vm are paths
    if load_fault:
        fault_params = load_yaml(load_fault)
    if load_root:        
       root_params = load_yaml(load_root)
    if load_vm:
       vm_params = load_yaml(load_vm)
```